deployment/deployer.go

```diff
@@ -101,6 +101,8 @@ type Deployer interface {
 	UpgradeCluster(ctx context.Context, clusterID string, CurrentImages string, NewImage string) error
 	AddNode(ctx context.Context, clusterID string) (string, error)
 	RemoveNode(ctx context.Context, clusterID string, nodeID string) error
+	FailOverNode(ctx context.Context, clusterID string, nodeID string, failOverType string, rebalance bool, allowUnsafe bool) error
+	FailOverRecoverNode(ctx context.Context, clusterID string, nodeID string, recoverType string, rebalance bool) error
```

Comment:
Shouldn't this just be called 'RecoverNode'?  Unless this performs a failover and then immediately recovers it?